TEDU-107 added segment section (#97)
mautic/user-documentation: docs/segments/manage_segments.rst

  .. image:: images/modify-segments.png
    :alt: Screenshot showing list of existing Segments.

5. Click Save and close.

.. vale off

Form submit action
==================

.. vale on

**Modify Contact’s Segment** is available as a submit action on Standalone Forms. With a Campaign Form, create a Campaign with a **Modify Contact’s Segment** action.

1. Click **Actions**.

2. From the **Add new submit action** menu, select **Modify Contact’s Segments.**

.. image:: images/form-submit-action.png
    :alt: Screenshot showing Form submit action.

3. Add a Title for the submit action and add a **Description** (optional).

4. Select the Segments you’d like to add the Contact to or remove the Contact from.

5. Click **Add**.

.. vale off

Points trigger
==============

.. vale on

Once a Contact has accrued an assigned number of Points, the system can add them to a Segment. This may be a Segment for your most engaged Contacts who become eligible for special offers, or a Segment your sales team reviews to find strong prospects.

.. image:: images/points-trigger.png
    :alt: Screenshot showing Points trigger.

1. In the **Points** section of the platform, select** Manage Triggers**.

> 2. Click **+New**.

Author: I agree since that's what we have in Mautic.

3. Give your trigger a **Name** and **Description**.

4. Enter the **Minimum number of Points** for adding a Contact to your Segment.

5. Decide if you’d like to add all Contacts with at least that number of Points to the Segment:
 

>   * If you only want to add new Contacts who reach the threshold to this Segment, select **No** (default).

Author: Using 'tip' markup will be the best because I have tried fixing the indentation and it's still same

  
  * To add all existing Contacts with at least a certain number of Points to the Segment, toggle the switch to **Yes**.

6. Click **Events**.

7. Click the menu for **Add an event** and select **Modify Contact’s Segments**.

8.  Add a **Title** for the event and a **Description** (optional).

9.  Select the Segments you’d like to add to or remove the Contact from.

10. Click **Add**.

11. Click **Save & Close**.

This accomplishes the same thing as creating a Dynamic Segment with a filter for Points, the operator ``greater than (or equal to)``, and the minimum number of Points. The difference is if you only want to add Contacts to a Segment who reach the point value after creating this trigger to a Segment, you can.

.. vale off

CSV upload
===========

.. vale on

> If you have a list outside of Mautic, saved as a UTF-8 encoded CSV file, you can upload the list directly into a Mautic Segment.

Reviewer: Maybe link here to the import section in managing contacts (may not be written yet!) - wonder if this documentation should sit there and we just link to it? What do you think? 

There's stuff like using the import in the background cron which are not relevant to add here (we just want to focus on the bit about how you get them into a segment), so I think it makes sense to point the user to the more in-depth docs.

Author: yeah, this section is in the contacts documentation which has not been written yet

5. Map the appropriate fields from the file. Remember to always map the unique identifier, so you don’t create duplicates.

6. Click **Import**.

If your file is larger than 1,000 rows, the system changes screens and informs you once the file has uploaded. After the upload is complete, click the Segments section of the platform to see the added Contacts.

.. vale off

Dynamic Segments
****************

.. vale on

.. image:: images/filtering.png
    :alt: Screenshot showing Segment filters.

Mautic moves Contacts into and out of dynamic Segments based on the filters applied to the Segment. As the data associated with the Contact updates, including Company associations and behaviors, Mautic updates Segment membership.

Segment filters
===============

.. image:: images/filter-list.png
    :width: 400
    :alt: Screenshot showing Segment Filters List.

1. Create a new Segment by clicking the **+New** button.

2. In the **Details** tab, add a **Name**, **Public name** and **Description** to your Segment.

3. Click the **Filters** tab to add filters.

4. Click the **Choose one**… menu and search for the field you’d like to Segment by. Listed below are three types of fields:
      

>   * Contact fields

Reviewer: Same issue here with the indentation - maybe it is 4 rather than 3 spaces?

Author: I tried 4 spaces and it is still same

  
    * Set Fields to **Available for Segments = Yes** in your Custom Field manager to display here.
  * Contact behavior and actions
  * Primary Company fields
    * Set Fields to **Available for Segments = Yes** in your Custom Field manager to appear here.
    * Contacts associates with multiple Companies, but Mautic adds them to Segments based on fields for the primary Company.

5. Add more filters, using the **And** and **Or** operators. An **Or** operator creates a new group of filters which can include And operators.

6. Click **Save and close**.

.. note:: 

>     The following filters rebuild immediately in real time, while other filters rebuild with the nightly refresh:

Reviewer: This is Acquia-specific so we should instead be replacing this with a note that segments are rebuilt according to how frequently you fire your cron jobs, and pointing people at the cron jobs docs about building segments.

    * All core fields
    * Custom fields added by the User
    * Category
    * Tags
    * Contact Email received
    * Hit URL
    * Owner ID

>     * The nightly Segment rebuild jobs for each Segment are randomized for six hours. Jobs are executed anytime between midnight to 6 AM with respect to the **Default timezone** set for the instance in **Settings > Configurations > System Settings**.

Reviewer: This is Acquia specific so needs to be removed. The two points below can remain.

.. vale off

Using Date Filters
==================

.. vale on

You can create dynamic Segments by using date filters.

.. note:: 
  
  The date format for values stored in the database is ``YYYY-MM-DD``. For example, December 11, 2020 is stored as 2020-12-11 and November 12, 2020 is stored as 2020-11-12. To update the display format for dates, go to **Settings > Configuration > System Settings > System Defaults**. However, this doesn’t alter the format in which dates are stored in the database.

.. vale off

Operators
=========

.. image:: images/operators.png
    :alt: Screenshot showing Operators.

.. vale on

You must ensure that you use the correct operator and time frame to build an effective Segment.

Once you’ve selected a date field as your filter, such as the default **Date last active** field or a custom **Birthday** field, you’ll have a list of operators to choose from:

* **Equals** - The value on the Contact record exactly matches the filter value.

* **Not Equal** - The value on the Contact record is any value that doesn’t match the filter value.

> * **Greater than** - The value on the Contact record is at a later date in time than X date. For example, **Greater than** today means anytime from tomorrow until the end of time.

Reviewer: In the front end it is a tiny bit confusing having the second bit of text in bold, especially when it lines up with the bold texts at the start of the line. I might suggest seeing how it looks using an inline code markup rather than bold, perhaps. See what you think!

* **Less than or equal** - The value on the Contact record is at an earlier date in time than X date. For example, **Less than or equal today** means anytime from the beginning of time until today.

* **Empty** - No value exists in the field on the Contact record.

* **Not empty** - A value exists in the field on the Contact record.

* **Like** - This operator isn't supported in date or date-time fields, and shouldn’t be used.

* **Not like** - This operator isn't supported in date or date-time fields, and shouldn’t be used.

* **Regexp** - Segment includes Contacts with values that match the specified regular expression pattern. If you aren't proficient with regular expression, don’t use this operator.

* **Not regexp** - Segment includes Contacts with values that don’t match the specified regular expression pattern. If you aren't proficient with regular expressions, don’t use this operator.

* **Starts with** - Segment includes Contacts whose field values begin with the specified numbers. These filter values should generally reference years (or years and months).

> For example, A value of 19 matches any Contacts whose field value has a year in the 1900&nbsp;s. A value of 200 matches Contacts with a year value between 2000 and 2009 and a value of 2020-11 matches Contacts with a field value in November 2020.

Reviewer: This looks a bit out of place as it has escaped the list. Please also fix the ascii characters.

* **Ends with** - Segment includes Contacts whose field values end with the specified numbers. These filter values should generally reference days (or months and days).

> For example, a value of 1 matches anyone whose field value is on the 1&nbsp;st, 21&nbsp;st, or 31&nbsp;st of any month but a value of 01 matches the 1&nbsp;st of a month. A value of 01-01 finds Contacts whose value is for January 1 of any year.

Reviewer: This looks odd, and also you have some ascii chars which need to be fixed.

* **Contains** - Segment includes Contacts with the specified filter value anywhere in the field value. 

.. image:: images/operators-2.png
    :alt: Screenshot showing Operators.

Once you have selected the field you can then choose the type of operation to perform. These vary depending on the way you wish to filter your Contacts.

.. vale off 

Matching part of a string
=========================

.. vale on

There are 5 filters you can use for matching part of a string - ``starts with``, ``ends with``, ``contains``, ``like`` and ``regexp``.
First three filters match strings as you enter it. ``like`` filter is for advanced Users - you can specify which type you want to use with ``%`` character:

* ``My string%`` is the same as ``starts with`` filter with ``My string`` value.

* ``%My string`` is the same as ``ends with`` filter with ``My string`` value.

* ``%My string%`` is the same as ``contains`` filter with ``My string`` value.

* ``My string`` is the same as ``contains`` filter with ``My string`` value.

A few notes for text filters:

> * ``starts with``, ``ends with``, ``contains`` filters were added later than the ``like`` one, so you can easily specify what you need now.

Reviewer: ```suggestion
* ``starts with``, ``ends with``, ``contains`` filters should be used rather than ``like`` as they are more specific, and therefore can be more performant.
```

> *  ``%`` character in the middle of the string has no special meaning. ``contains`` filter with ``my % string`` will search for a string with ``%`` in the middle. The same is TRUE for ``like`` filter with ``%my % string%`` value. There is no need for escaping this character.

Reviewer: ```suggestion
*  A ``%`` character in the middle of the string has no special meaning. A ``contains`` filter with ``my % string`` will search for a string with ``%`` in the middle. The same is TRUE for a ``like`` filter with ``%my % string%`` value. There is no need to escape this character.
```

> * Mautic searches for ``%`` character in a value for ``like`` filter and no modification is performed if at least 1 ``%`` is found.

Reviewer: ```suggestion
* Mautic searches for the ``%`` character in a value for the ``like`` filter and no modification is performed if at least one ``%`` is found.
```

You can use regular expressions in a ``regexp`` filter. Mautic recognises all common operators like ``|`` for OR (``first string|second string``), character sets (``[0-9]``, ``[a-z0-9]`` etc.), repetitions (``+``, ``*``, ``?``) and more. 
You have to escape special characters with ``\`` if you want to use them as matching character. 
Learn more about :xref:`Regex`. 

.. note:: 

  MySQL (and Mautic) uses POSIX regex, which could behave differently from other types of Regex.

.. vale off

Date options
============

.. vale on

Date filters allow you to choose a date via DatePicker:

.. image:: images/date-filters.png
    :alt: Screenshot showing DatePicker.

However, you can specify much more here. Mautic recognizes relative formats too (these string aren't translatable):

* ``+1 day`` (you can also use ``1 day``)
* ``-2 days`` (you can also use ``2 days ago``)
* ``+1 week`` / ``-2 weeks`` / ``3 weeks ago``

> * ``+5 months`` / ``-6 months`` / ``7months ago``

Reviewer: ```suggestion
* ``+5 months`` / ``-6 months`` / ``7 months ago``
```

* ``birthday`` / ``anniversary``
* ``birthday -7 days`` / ``anniversary -7 days``
* ``today`` / ``tomorrow`` / ``yesterday``
* ``this week`` / ``last week`` / ``next week``
* ``this month`` / ``last month`` / ``next month``
* ``this year`` / ``last year`` / ``next year``

Example (Consider that today is ``2022-03-05``):

* ``Date identified equals last week`` returns all Contacts identified between 2022-03-01 and 2022-03-07 (Monday - Sunday).
* ``Date identified less than last week`` returns all Contacts identified before 2022-02-22.
* ``Date identified equals last month`` returns all Contacts identified between 2022-02-01 and 2022-02-28.
* ``Date identified greater or equal last year`` returns all Contacts identified 2021-01-01 and after.
* ``Date identified greater than last year`` returns all Contacts identified after 2021-12-31.
* ``Custom Contact date field equal birthday -1 day`` returns all Contacts identified every year on 03-04 (4th march).
* ``Custom Contact date field equal anniversary -1 month`` returns all Contacts identified every year on 02-04 (4th february)

> Once you have created your Segment, any applicable Contact automatically add through the execution of a cron job. This is the essence of Segments.

Reviewer: ```suggestion
Once you have created your Segment, any applicable Contact will be automatically added through the execution of a cron job. This is the essence of Segments.
```
Maybe consider linking to the relevant cron job documentation.

To keep the Segments current, create a cron job that executes the following command at the desired interval:

> ``php /path/to/mautic/bin/console mautic:Segments:update``

Reviewer: ```suggestion
``php /path/to/mautic/bin/console mautic:segments:update``
```

Reviewer: I think we should instead link to the docs, personally, rather than duplicate here.

> Through the execution of that command, it adds Contacts that match the filters and removes Contacts that no longer match. Any Contacts that were manually added will remain part of the list regardless of filters.

Reviewer: ```suggestion
Through the execution of that command, Mautic adds Contacts that match the filters and removes Contacts that no longer match. Any Contacts that were manually added will remain part of the list regardless of filters.
```

.. vale off

Delete all Contacts in a Segment
********************************

.. vale on

> Filter the Contacts in the Segment. The batch delete action in the Contact table allows deletion of up to 100 Contacts at one time. 

Reviewer: What is the purpose of having the 'Filter the Contacts in a Segment' here?

If it is an instruction (eg that they need to first create the segment) we should be clearer here.

Maybe:

'It is possible to delete all contacts in a segment manually rather than with a Campaign action. To do this, first create a segment with filters that will pick up all the Contacts you want to delete.'

> This is a performance precaution since deleting more Contacts at one time could cause issues. Hundreds of Contacts can use this feature.

Reviewer: ```suggestion
This is a performance precaution since deleting more Contacts at one time could cause issues. You can, however, delete multiple batches of 100 contacts to delete larger lists.
```

1. Select the checkboxes next to those Contacts.

2. Click the green arrow which appears at the top of the column.

3. Select **Delete Selected** from the list.

> 4. Choose the **Contacts** you’d like to delete in a Segment.

Reviewer: You already did this in step 1, I don't think it needs to be repeated here?

5. Click **Delete**.

.. image:: images/select-delete.png
    :width: 200
    :alt: Screenshot showing Deleting all Contacts in a Segment.

> But deleting thousands of Contacts this way in one Segment becomes a tedious task. Luckily, there is a trick how to let the background workers do the job for you.

Reviewer: ```suggestion
Deleting thousands of Contacts this way in one Segment becomes a tedious task. Luckily, there is a trick how to let the background workers do the job for you.
```

1. Create a simple Campaign which has the Segment as the source

2. Use the :ref:`Delete contact action<using the campaign builder>`.

> This way the ``mautic:campaign:update`` and ``mautic:campaign:trigger`` commands deletes all the Contacts in the Segment. As well as all the Contacts which adds to the Segment in the future. 

Reviewer: ```suggestion
This way the ``mautic:campaign:update`` and ``mautic:campaign:trigger`` commands delete all the Contacts in the Segment, and all the Contacts added to the Segment in the future. 
```

It's all done automatically in the background.

> It's necessary to configure the cron jobs. Deleted Contacts can't be recovered.

Reviewer: Link to the relevant section of the cron jobs documentation.

Personally I would put the part about the contacts not being recoverable in a big red warning box and advise folks to use this feature very carefully!
